rainbow-me/rainbowkit
# fix: prevent `eth_getBalance` call in `<ConnectButton />` when `showBalance` is false for either large or small screen

## Changes
- Fixed a bug where `<ConnectButton />` component was calling the "eth_getBalance" RPC call even when `showBalance` was set to `false` for either large or small screen.

- Fixed a bug in the `<AccountModal />` component so it now calls "eth_getBalance" only when the modal is opened, not automatically during mounting.

## What to test
- Make sure the "eth_getBalance" call is not being made unless the `showBalance` prop in the `<ConnectButton />` component is true for one of the screen sizes (large or small)
- Make sure the `<AccountModal />` component is not calling "eth_getBalance" during mounting.
